source/common/router/header_parser_utils.cc

```diff
@@ -48,11 +48,18 @@ std::string HeaderParser::translateMetadataFormat(const std::string& header_valu
           "Header formatter: JSON format of {} parameters has been obsoleted. Use colon format: {}",
           metadata_type.as_string() + "_METADATA", new_format.c_str());
 
-      re2::RE2::Replace(&new_header_value, re, new_format);
+      if (!re2::RE2::Replace(&new_header_value, re, new_format)) {
```

Comment:
I looked at the corpus files and still do not fully grasp how it is that regex finds the pattern but fails in replacing it.

Reply:
I've dug into this a bit deeper. In `RE2::Replace` backslash-escaped digits are used to refer to specific matches.
If the replaced string had that, it could keep the original (wrong) pattern after the replacement, causing an infinite loop.
I've updated the code to use `RE2::Match` and `absl::StrReplaceAll` instead.

Reply:
Thanks for explaining it.